lib/modules/manager/pip_requirements/artifacts.ts

```diff
@@ -25,10 +28,13 @@ export async function updateArtifacts({
       .split(newlineRegex)
       .map((line) => line.trim());
     for (const dep of updatedDeps) {
+      const depName = dep.depName!;
       const hashLine = lines.find(
         (line) =>
           // TODO: types (#7154)
-          line.startsWith(`${dep.depName!}==`) && line.includes('--hash=')
+          line.startsWith(depName) &&
+          extrasAndEqualsRegex.test(line.substring(depName.length)) &&
+          line.includes('--hash=')
       );
       if (hashLine) {
         const depConstraint = hashLine.split(' ')[0];
```

Comment:
then here we wouldnt just split, instead you can have capture naming groups and put the group name here to understand what we are really picking up

Reply:
Understood. All I need to know is how to proceed with escaping the dependency name inside the regex. Then I can push a change that extracts all the necessary bits during the regex matching.

Reply:
There is a function to escape depname its used on depnames all over the code ill check it tomorrow

Reply:
I already found the escape function. e8eccbf48099da83b0b6cc15b441160e363fc65e uses a named subgroup to extract the dependency constraint.

Reply:
yup that's the one.